apps/archive/commands.py

```diff
@@ -80,19 +81,26 @@ def _remove_expired_items(self, expiry_datetime, log_msg):
             item.setdefault('expiry', expiry_datetime)
             item.setdefault('unique_name', '')
             expiry_msg = log_msg_format.format(**item)
-            logger.info('{} Processing expired item. {}'.format(log_msg, expiry_msg))
+            logger.info('{} Processing expired item. {}'.format(self.log_msg, expiry_msg))
 
             processed_items = set()
             if item_id not in items_to_be_archived and self._can_remove_item(item, processed_items):
                 # item can be archived and removed from the database
-                logger.info('{} Removing item. {}'.format(log_msg, expiry_msg))
-                logger.info('{} Items to be removed. {}'.format(log_msg, processed_items))
+                logger.info('{} Removing item. {}'.format(self.log_msg, expiry_msg))
+                logger.info('{} Items to be removed. {}'.format(self.log_msg, processed_items))
                 items_to_be_archived = items_to_be_archived | processed_items
 
+        items_to_expire = items_to_be_archived | set([item.get(config.ID_FIELD) for item in spiked_killed_items
+                                                      if item.get(ITEM_STATE) == CONTENT_STATE.KILLED])
+
+        if not self.check_if_items_imported_to_legal_archive(list(items_to_expire)):
+            logger.exception('{} CANNOT EXPIRE ITEMS AS ITEMS ARE NOT MOVE TO LEGAL ARCHIVE.'.format(self.log_msg))
+            return
```

Comment:
Agree on that it will no remove items if they are in legal archive. I didn't want to complicate things, due to package/broadcast dependencies . Hence, I am checking for all items for now.